compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1393,7 +1392,13 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
             if should_encode_fn_sig(def_kind) {
                 record!(self.tables.fn_sig[def_id] <- tcx.fn_sig(def_id));
             }
-            if should_encode_generics(def_kind) {
+            // FIXME: Some anonymous constants produced by `#[rustc_legacy_const_generics]`
```

Comment:
Should we give them a dummy hir node to avoid those ices?

Reply:
Maybe, I just didn't want to get into that in this PR, https://github.com/rust-lang/rust/pull/118188 is not blocked on it.

I tried to look what exactly happens with `rustc_legacy_const_generics`, but it's still not clear to me.
There seems to be some mismatch between what def collector and AST lowering do.